lib/common/client/middleware/response/snakecase.rb

```diff
@@ -24,9 +24,18 @@ def underscore_symbolize(hash)
             hash.deep_transform_keys { |k| k.underscore.to_sym }
           end
         end
+
+        class SnakecaseString < Snakecase
+          private
+
+          def underscore_symbolize(hash)
+            hash.deep_transform_keys(&:underscore)
```

Comment:
Agree with @saneshark on the naming not making sense anymore. Could the snakecase middleware parse the JSON objects into hashes with indifferent access?